src/SprykerSdk/Zed/AppSdk/Business/Validator/AsyncApi/AsyncApiValidator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Zed\AppSdk\Business\Validator\AsyncApi;
+
+use Behat\Gherkin\Exception\ParserException;
+use Exception;
+use Generated\Shared\Transfer\MessageTransfer;
+use Generated\Shared\Transfer\ValidateRequestTransfer;
+use Generated\Shared\Transfer\ValidateResponseTransfer;
+use SprykerSdk\Zed\AppSdk\Business\Validator\AbstractValidator;
+use Symfony\Component\Yaml\Yaml;
+
+class AsyncApiValidator extends AbstractValidator
+{
+    /**
+     * @param \Generated\Shared\Transfer\ValidateRequestTransfer $validateRequestTransfer
+     * @param \Generated\Shared\Transfer\ValidateResponseTransfer|null $validateResponseTransfer
+     *
+     * @throws \Behat\Gherkin\Exception\ParserException
+     *
+     * @return \Generated\Shared\Transfer\ValidateResponseTransfer
+     */
+    public function validate(
+        ValidateRequestTransfer $validateRequestTransfer,
+        ?ValidateResponseTransfer $validateResponseTransfer = null
+    ): ValidateResponseTransfer {
+        $validateResponseTransfer ??= new ValidateResponseTransfer();
+
+        if (!$this->finder->hasFiles($validateRequestTransfer->getAsyncApiPathOrFail())) {
```

Comment:
Since you are using the file name in many places I would get it once before this line

`$asyncApiFile = $validateRequestTransfer->getAsyncApiFileOrFail();`